crates/rbuilder/src/backtest/fetch/mod.rs

```diff
@@ -176,29 +180,33 @@ impl HistoricalDataFetcher {
                 async move {
                     let mut all_nonces_failed = true;
                     for nonce in nonces {
-                        let onchain_nonce = {
-                            let mut nonce_cache = nonce_cache.lock().await;
+                        let mut res_onchain_nonce = 0_u64;
```

Comment:
Would it make more sense to us an `Option<u64>` instead? Imagine that the nonce_cache is initialized and has a value of `0` for this address already. Then we hit the `if res_onchain_nonce == 0` codepath and repeat the nonce call path even tho we already know the appropriate nonce value. It's not the worst thing in the world, but there is perhaps some more optimization you could do!